nestedtensor/nested/masking.py

```diff
@@ -25,45 +22,41 @@ def nested_tensor_from_tensor_mask(tensor, mask, nested_dim=None):
     if not mask.dim() > 0:
         raise RuntimeError("Mask has to have dimention > 0")
 
-    # TODO: Need to define empty-list and 0 numel semantics
-    # TODO: Test edge-cases and kwargs
-    def _merge(tensors, nested_dim=None):
-        assert len(tensors) > 0
+    if nested_dim is not None and mask.dim() < nested_dim:
+        raise ValueError("Mask dimension ({0}) is too small to construct nested tensor with nested dimension of ({1})".format(mask.dim(), nested_dim))
+
+    def create_nested_tensor(tensors, nested_dim=None):
+        assert len(tensors) > 0, "Can't merge empty list of tensors. Please, report this error."
 
-        def _mergable_tensors(tensors, nested_dim):
+        def should_create_nested_tensor(tensors, nested_dim):
             if nested_dim is None:
-                size0 = tensors[0].size()
-                compliant = all(torch.is_tensor(t) for t in tensors)
-                compliant = all(size0 == t.size()
-                                for t in tensors) and compliant
-                return compliant
+                size = tensors[0].size()
+                return not all(torch.is_tensor(t) and size == t.size() for t in tensors)
```

Comment:
You could do not torch.is_tensor(t) or size != t.size() and use "any". That'll be more efficient because it can stop earlier. That should be more efficient as well if I'm not mistaken for what that usual case that you're expecting here is.